pandas/core/frame.py

```diff
@@ -7015,7 +7015,7 @@ def append(self, other, ignore_index=False, verify_integrity=False, sort=None):
             other = other._convert(datetime=True, timedelta=True)
             if not self.columns.equals(combined_columns):
                 self = self.reindex(columns=combined_columns)
-        elif isinstance(other, list) and not isinstance(other[0], DataFrame):
+        elif isinstance(other, list) and other and not isinstance(other[0], DataFrame):
```

Comment:
instead of this, do
```
elif isinstancde(other, list):
   ....
```

Reply:
do you mean to have a nested structure as below?
```
elif isinstance(other, list):
    if # check empty
    elif # check first item is not a DataFrame
```

Reply:
@jreback The reason added that part was because that is caught at https://github.com/asishm/pandas/blob/append_empty_bugfix/pandas/core/frame.py#L7025.